karapace/serialization.py

```diff
@@ -166,7 +166,7 @@ async def get_schema_for_subject(self, subject: str) -> TypedSchema:
         assert self.registry_client, "must not call this method after the object is closed."
         schema_id, schema = await self.registry_client.get_latest_schema(subject)
         async with self.state_lock:
-            schema_ser = schema.__str__()
+            schema_ser = schema.__str__()  # pylint: disable=unnecessary-dunder-call
```

Comment:
`str(schema)`

Reply:
I addressed all of these, plus the one suggesting `enumerate()`. I'm a bit hesitant to addressing the two `consider-using-with` in this same PR, but if you think that makes sense I can do so.